box.007

```diff
@@ -1,84 +1,81 @@
 my text = [
-    "This is random text",
+    "This is a random text",
     "that will be boxed with this function.",
     "I have set it to be 3"
 ];
 
 boxify(text, 4);
 
-func boxify(strings_to_box, w_on_sides) {
+func boxify(stringsToBox,wOnSides) {
 
     my index = 0;
-    my longest_str_len = 0;
+    my longestStrLen = 0;
 
-    for(strings_to_box) -> string {
-        my str_len = string.chars();
-        if(str_len > longest_str_len) {
-            longest_str_len = str_len;
+    for stringsToBox -> string {
+        my strLength = string.chars();
+        if strLength > longestStrLen {
+            longestStrLen = strLength;
         }
     }
```

Comment:
On an unrelated note: should definitely have a built-in `.max` method in 007 that can take a (named?) argument specifying sort order. Prior art: [Python](https://docs.python.org/3.8/library/functions.html#max), [Perl 6](https://docs.perl6.org/routine/max). Then the above code could be just:

    my longestStrLen = stringsToBox.map(sub (s) { s.chars() }).max();

Reply:
007 used to have `min` and `max` (as built-in subs), but [they were removed](https://github.com/masak/007/commit/429fb40c23d3ccaf2a9878ebfcc581e9b9df5517) because they _one_ place they were used in `examples/` could be re-written as a pretty clear `if` statement instead.

I think we'll revive `min` and `max` at some point, as built-in subs taking a vararg and a named `key` argument, but (a) 007 doesn't really have varargs exposed in the language yet, (b) we're just about to get named arguments, (c) the easiest way to write anonymous functions is still a bit clumsy (but see [#202](https://github.com/masak/007/issues/202) and [#215](https://github.com/masak/007/issues/215) for some future syntactic relief), and (d) the need for these aren't overwhelming yet, since the ecosystem is so small. I'm guessing these will be used somewhere in the bootstrapped compiler too, though.

Oh, and `sort()` should totally have a `key` also. I'd better create a 007 issue about all of this.

Reply:
[Done](https://github.com/masak/007/issues/437).